lib/elixir/test/elixir/io_test.exs

```diff
@@ -118,6 +118,19 @@ defmodule IOTest do
     assert capture_io(fn -> IO.puts(13) end) == "13\n"
   end
 
+  test "warn with chardata" do
+    assert capture_io(:stderr, fn -> IO.warn("hello") end) =~ "warning: hello\n    (ex_unit) lib/ex_unit"
+    assert capture_io(:stderr, fn -> IO.warn('hello') end) =~ "warning: hello\n    (ex_unit) lib/ex_unit"
+    assert capture_io(:stderr, fn -> IO.warn(:hello) end) =~ "warning: hello\n    (ex_unit) lib/ex_unit"
+    assert capture_io(:stderr, fn -> IO.warn(13) end) =~ "warning: 13\n    (ex_unit) lib/ex_unit"
+    assert capture_io(:stderr, fn -> IO.warn("hello", []) end) == "warning: hello\n\n"
```

Comment:
@lpil we should probably get rid of it otherwise someone will certainly report it as an error :)